src/Core/Charset.cs

```diff
@@ -0,0 +1,137 @@
+/* ***** BEGIN LICENSE BLOCK *****
+ * Version: MPL 1.1/GPL 2.0/LGPL 2.1
+ *
+ * The contents of this file are subject to the Mozilla Public License Version
+ * 1.1 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * http://www.mozilla.org/MPL/
+ *
+ * Software distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
+ * for the specific language governing rights and limitations under the
+ * License.
+ *
+ * Contributor(s):
+ *          UTF-Unknown Contributors (2019)
+ * 
+ * Alternatively, the contents of this file may be used under the terms of
+ * either the GNU General Public License Version 2 or later (the "GPL"), or
+ * the GNU Lesser General Public License Version 2.1 or later (the "LGPL"),
+ * in which case the provisions of the GPL or the LGPL are applicable instead
+ * of those above. If you wish to allow use of your version of this file only
+ * under the terms of either the GPL or the LGPL, and not to allow others to
+ * use your version of this file under the terms of the MPL, indicate your
+ * decision by deleting the provisions above and replace them with the notice
+ * and other provisions required by the GPL or the LGPL. If you do not delete
+ * the provisions above, a recipient may use your version of this file under
+ * the terms of any one of the MPL, the GPL or the LGPL.
+ *
+ * ***** END LICENSE BLOCK ***** */
+
+namespace UtfUnknown.Core
+{
+    /// <summary>
+    /// charset helper
+    /// </summary>
+    /// <remarks>Based on https://github.com/dotnet/corefx/blob/cf28b7896a762f71c990a5896a160a4138d833c9/src/System.Text.Encoding.CodePages/src/System/Text/EncodingTable.Data.cs</remarks>
+    internal static class Charset
+    {
+        internal const string ASCII = "ascii";
+
+        internal const string UTF8 = "utf-8";
+
+        internal const string UTF16_LE = "utf-16le";
+        
+        internal const string UTF16_BE = "utf-16be";
+        
+        internal const string UTF32_LE = "utf-32le";
+        
+        internal const string UTF32_BE = "utf-32be";
+
+        internal const string X_ISO_10646_UCS_4_3412 = "X-ISO-10646-UCS-4-3412"; // TODO: not supported?
```

Comment:
please move these comments to XML comments (summary or remarks).

Then it's visible for other places and easy to refactor

Reply:
Ready ;)